java-checks/src/main/java/org/sonar/java/checks/naming/BadMethodNameCheck.java

```diff
@@ -57,16 +57,18 @@ public void setContext(JavaFileScannerContext context) {
     super.setContext(context);
   }
 
-
   @Override
   public void visitNode(Tree tree) {
     MethodTree methodTree = (MethodTree) tree;
-    if (isNotOverriden(methodTree) && !pattern.matcher(methodTree.simpleName().name()).matches()) {
+    if (!isExcluded(methodTree) && !pattern.matcher(methodTree.simpleName().name()).matches()) {
       reportIssue(methodTree.simpleName(), "Rename this method name to match the regular expression '" + format + "'.");
     }
   }
 
-  private static boolean isNotOverriden(MethodTree methodTree) {
-    return Boolean.FALSE.equals(methodTree.isOverriding());
+  private boolean isExcluded(MethodTree methodTree) {
+    return !Boolean.FALSE.equals(methodTree.isOverriding()) || methodTree.modifiers().annotations().stream().anyMatch(
+      // use simple name, so we have no FP on missing semantics
+      it -> "AttributeDefinition".equals(it.annotationType().symbolType().name())
+    );
```

Comment:
How about extracting the annotation check to its own method for readability?
```suggestion
  private static boolean isExcluded(MethodTree methodTree) {
    return !Boolean.FALSE.equals(methodTree.isOverriding()) ||
      isExcludedByAnnotation(methodTree);
  }
  
  private static boolean isExcludedByAnnotation(methodTree) {
    return methodTree.modifiers().annotations().stream().anyMatch(
      // use simple name, so we have no FP on missing semantics
      it -> "AttributeDefinition".equals(it.annotationType().symbolType().name())
    );
```